### ISSUE TITLE
Problem extracting archives with hard links

### ISSUE THREAD

**Issue by petemoore** [mholt/archiver#213]
## What version of the package or command are you using?
3.3.0

## What are you trying to do?
Extract https://github.com/git-for-windows/git/releases/download/v2.11.0.windows.3/Git-2.11.0.3-32-bit.tar.bz2

## What steps did you take?
Built arc from `v3.3.0` branch, then ran `arc unarchive Git-2.11.0.3-32-bit.tar.bz2`.

## What did you expect to happen, and what actually happened instead?
I hoped for a successful extraction, but got the errors [here](https://github.com/mholt/archiver/pull/171#issuecomment-588129903).

## How do you think this should be fixed?
Maybe the code from #171 needs to be merged to master?

## Please link to any related issues, pull requests, and/or discussion
#171 #152 #153 
I created a new issue, since I'm not able to reopen the ones above. I hope that is ok.

## Bonus: What do you use archiver for, and do you find it useful?
Mozilla uses it in generic-worker, which is part of taskcluster, which is what Mozilla uses for build and release automation for almost all of its products, including the firefox browser.

**Comment by mholt** [comment]
Can you open a pull request with the fix to master? That will make things faster.

**Comment by petemoore** [comment]
> Can you open a pull request with the fix to master? That will make things faster. I'm swamped with other things right now, unfortunately.

Apologies, it did end up on master, under a different commit id (f426e4246a0ca99631906e94a3ec803254c99935  -> 9ea51e7370bcb0ca5aa6318b481c2fd51697db9f). So there is not anything to merge after all.

So this suggests that the original issue (#152) wasn't fixed by the change (#171).

I'm sorry about this as [you did originally ask me to test the fix](https://github.com/mholt/archiver/pull/153#issuecomment-553973378), and I never got around to it until recently. That was an oversight on my part.

Please note though, #152 and this issue contain the same reproducible test case, so hopefully it should be straightforward to see if any future fix works, by testing whether the [cited archive](https://github.com/git-for-windows/git/releases/download/v2.11.0.windows.3/Git-2.11.0.3-32-bit.tar.bz2) can be successfully extracted on either macOS or Windows.

Many thanks!

**Comment by petemoore** [comment]
> Please note though, #152 and this issue contain the same reproducible test case, so hopefully it should be straightforward to see if any future fix works, by testing whether the [cited archive](https://github.com/git-for-windows/git/releases/download/v2.11.0.windows.3/Git-2.11.0.3-32-bit.tar.bz2) can be successfully extracted on either macOS or Windows.

This seems to be fixed now. Not sure why there isn't a v3.5.1 binary release, but using the v3.5.0 the archive now extracts correctly:

```bash
$ curl -L 'https://github.com/mholt/archiver/releases/download/v3.5.0/arc_3.5.0_mac_amd64' > arc
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   656  100   656    0     0   2877      0 --:--:-- --:--:-- --:--:--  2877
100 6283k  100 6283k    0     0  4491k      0  0:00:01  0:00:01 --:--:-- 6545k
$ chmod a+x arc
$ curl -L https://github.com/git-for-windows/git/releases/download/v2.11.0.windows.3/Git-2.11.0.3-32-bit.tar.bz2 > git.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   664  100   664    0     0   3458      0 --:--:-- --:--:-- --:--:--  3458
100 76.0M  100 76.0M    0     0  16.9M      0  0:00:04  0:00:04 --:--:-- 19.1M
$ ./arc unarchive git.tar.bz2 git2
$ echo $?
0
```

I haven't tried with v4 of archiver.
